main.cpp

```diff
@@ -78,81 +77,102 @@ void RunTests(string _seedsetFile, string _resultsFile)
     getline(seedsetInput, line);
 
     std::istringstream iss(line);
-    int seeds, width, height, colourCount, totalMoves;
+    int seedCount, width, height, colourCount, maxMoves;
     getline(iss, line, ',');
-    seeds = stoi(line);
+    seedCount = stoi(line);
     getline(iss, line, ',');
     width = stoi(line);
     getline(iss, line, ',');
     height = stoi(line);
     getline(iss, line, ',');
     colourCount = stoi(line);
     getline(iss, line, ',');
-    totalMoves = stoi(line);
+    maxMoves = stoi(line);
 
     int sum = 0;
+    int errorCount = 0;
     string str = "";
-    while (getline(seedsetInput, line))
+    for (int i = 0; i < seedCount; i++)
     {
+        getline(seedsetInput, line);
+        if (line.empty())
+        {
+            cout << "ERROR: The test found an empty line in the given seedset.\n";
+            return;
+        }
         int s = stoi(line);
+
         Player *p = new BasicAI();
-        Game game(p, width, height, colourCount, totalMoves, s);
+        Game game(p, width, height, colourCount, maxMoves, s);
         int moves = game.play();
-        sum += moves;
-        str += std::to_string(moves);
-        str += "\n";
+
+        if (moves == -1)
+        {
+            errorCount++;
+            str += "GAME " + std::to_string(i + 1) + ": **TOO MANY MOVES**\n";
+        }
+        else
+        {
+            sum += moves;
+            str += "GAME " + std::to_string(i + 1) + ": " + std::to_string(moves) + "\n";
+        }
     }
 
     std::ofstream out(_resultsFile);
-    out << sum << "\n";
+    out << "TOTAL MOVES: " << std::to_string(sum) << "\n";
+    out << "AVG. MOVES: " << std::to_string(sum / seedCount) << "\n";
```

Comment:
Why was this changed? Adding needless text to the save file only makes reading more difficult.

By the way, we should probably keep the results file to just the individual board results. All these others stats should be calculated later in a separate program.